### PR TITLE
Use trusted bundle with root CAs for multi-tenant manager pods

### PR DESCRIPTION
**Pull request by caseydavenport**
## Description

Manager pods need to use a trusted bundle that includes the system
certificates, but other pods do not. 

This PR introduces the ability in a multi-tenant system to deploy both
types of bundle, and enables individiual components to decide which of
the two they need to load.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/tls/certificatemanagement/interface.go [hunk 1]
```diff
@@ -23,12 +23,19 @@ import (
...
 	TrustedCertBundleMountPath        = "/etc/pki/tls/certs/tigera-ca-bundle.crt"
 	TrustedCertBundleMountPathWindows = "c:/etc/pki/tls/certs/tigera-ca-bundle.crt"
+
+	// TrustedCertConfigMapName is the name of the trusted certificate bundle ConfigMap. This value is used
+	// for all single-tenant trusted bundles, as well as multi-tenant trusted bundles that do not include public CAs.
+	TrustedCertConfigMapName = "tigera-ca-bundle"
+
+	// TrustedCertConfigMapNamePublic is the name of the trusted certificate bundle ConfigMap that includes public CAs, used
+	// only in multi-tenant environments as a single namespace requires both a trusted bundle with public CAs as well as one without.
+	TrustedCertConfigMapNamePublic = "tigera-ca-bundle-public"
```
Comment (caseydavenport) [inline pkg/tls/certificatemanagement/interface.go:38]: Probably want to rename this to `tigera-ca-bundle-with-system-roots`

Comment (asincu) [reply pkg/tls/certificatemanagement/interface.go:38]: I see that for a single tenant, we should use the same config map `tigera-ca-bundle`, but with the following fields
- `tigera-ca-bundle.crt` holds certificates generated by tigera ca
- `ca-bundle.crt`holds public certificates

Why can't we use the same mechanism ?

Comment (caseydavenport) [reply pkg/tls/certificatemanagement/interface.go:38]: We don't want to mount the public certificates into ever pod, and this approach just seemed the most in line with how we currently manage these for enterprise. I think it's a bit easier to grasp conceptually.

**Review by asincu — COMMENTED** [review]
Looks good! I think some static checks are failing, but the correct bundle is mounted now.

**Review by asincu — APPROVED** [review]

**Pull request merged**